src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaReviewSchedulePersistenceAdapter.kt

```diff
@@ -40,8 +41,21 @@ class JpaReviewSchedulePersistenceAdapter(
   }
 
   @Transactional(readOnly = true)
-  override fun getReviewSchedule(prisonNumber: String): ReviewSchedule? =
-    reviewScheduleRepository.findByPrisonNumber(prisonNumber)?.let {
-      reviewScheduleEntityMapper.fromEntityToDomain(it)
+  override fun getActiveReviewSchedule(prisonNumber: String): ReviewSchedule? =
+    with(
```

Comment:
Lets say a prisoner has been in prison for years. then this will bring back lots of records and then filter on the active one. 

Probably better to create a HQL query that simply returns a single result using a where status != ReviewScheduleStatus.COMPLETED which would return a NonUniqueResultException for the case where the database has got dodgy data and do the filtering in the DB which will perform better